api/src/controllers/feature-flag.controller.ts

```diff
@@ -0,0 +1,115 @@
+import {
+  Body,
+  Controller,
+  Delete,
+  Get,
+  Param,
+  ParseUUIDPipe,
+  Post,
+  Put,
+  UseGuards,
+  UsePipes,
+  ValidationPipe,
+} from '@nestjs/common';
+import {
+  ApiExtraModels,
+  ApiOkResponse,
+  ApiOperation,
+  ApiTags,
+} from '@nestjs/swagger';
+import { FeatureFlagService } from '../services/feature-flag.service';
+import { FeatureFlag } from '../dtos/feature-flags/feature-flag.dto';
+import { FeatureFlagAssociate } from '../dtos/feature-flags/feature-flag-associate.dto';
+import { FeatureFlagCreate } from '../dtos/feature-flags/feature-flag-create.dto';
+import { FeatureFlagUpdate } from '../dtos/feature-flags/feature-flag-update.dto';
+import { defaultValidationPipeOptions } from '../utilities/default-validation-pipe-options';
+import { IdDTO } from '../dtos/shared/id.dto';
+import { SuccessDTO } from '../dtos/shared/success.dto';
+import { PermissionTypeDecorator } from '../decorators/permission-type.decorator';
+import { OptionalAuthGuard } from '../guards/optional.guard';
+import { PermissionGuard } from '../guards/permission.guard';
+import { ApiKeyGuard } from '../guards/api-key.guard';
+
+@Controller('featureFlags')
+@ApiTags('featureFlags')
+@UsePipes(new ValidationPipe(defaultValidationPipeOptions))
+@ApiExtraModels(
+  FeatureFlagAssociate,
+  FeatureFlagCreate,
+  FeatureFlagUpdate,
+  IdDTO,
+)
+@PermissionTypeDecorator('featureFlags')
+@UseGuards(OptionalAuthGuard, PermissionGuard)
+export class FeatureFlagController {
+  constructor(private readonly featureFlagService: FeatureFlagService) {}
+
+  @Get()
+  @ApiOperation({ summary: 'List of feature flags', operationId: 'list' })
+  @ApiOkResponse({ type: FeatureFlag, isArray: true })
+  async list(): Promise<FeatureFlag[]> {
+    return await this.featureFlagService.list();
+  }
+
+  @Post()
+  @ApiOperation({
+    summary: 'Create a feature flag',
+    operationId: 'create',
+  })
+  @ApiOkResponse({ type: FeatureFlag })
+  @UseGuards(ApiKeyGuard)
```

Comment:
since this requires an admin to making these changes wondering if the whole controller should have the ApiKeyGuard applied instead of on basically all the endpoints